File: AtCoder/ACP/BootCamp4b/Hard100/python/D_Gathering_Children.py
S = input()
N = len(S)
C = [0] * N
i = 0
while i < N:
    cnt = 0
    while i < N and S[i] == 'R':
        i += 1
        cnt += 1
    C[i - 1] += cnt // 2 + int(cnt % 2 != 0)
    C[i] += cnt // 2
    while i < N and S[i] == 'L':
        i += 1
i = N - 1
while i >= 0:
    cnt = 0
    while i >= 0 and S[i] == 'L':
        i -= 1
        cnt += 1
    C[i + 1] += cnt // 2 + int(cnt % 2 != 0)
    C[i] += cnt // 2
    while i >= 0 and S[i] == 'R':
        i -= 1
for c in C:
    print(c)

File: AtCoder/ACP/BootCamp4b/Hard100_2/cpp/D_Gathering_Children.cpp
#include <bits/stdc++.h>
using namespace std;

string S;

void solve() {
  int N = S.size();
  vector<int> C(N);
  int cnt = 0;
  for (int i = 0; i < N; ++i) {
    if (S[i] == 'R') {
      ++cnt;
    } else {
      C[i] += cnt / 2;
      if (i - 1 >= 0) {
        C[i - 1] += (cnt + 2 - 1) / 2;
      }
      cnt = 0;
    }
  }
  cnt = 0;
  for (int i = N - 1; i >= 0; --i) {
    if (S[i] == 'L') {
      ++cnt;
    } else {
      C[i] += cnt / 2;
      if (i + 1 <= N - 1) {
        C[i + 1] += (cnt + 2 - 1) / 2;
      }
      cnt = 0;
    }
  }
  for (int i = 0; i < N; ++i) {
    printf("%d%c", C[i], i == N - 1 ? '\n' : ' ');
  }
}

int main() {
  cin >> S;
  solve();
}